Remotion/Web/Development.WebTesting/WebTestSetUpFixtureHelper.cs

```diff
@@ -144,31 +144,37 @@ private void VerifyWebApplicationStarted (Uri webApplicationRoot, TimeSpan appli
 
       var stopwatch = Stopwatch.StartNew();
 
-#pragma warning disable SYSLIB0014
-      var webRequest = (HttpWebRequest)HttpWebRequest.Create(resolvedUri); // TODO RM-8492: Replace with HttpClient
-#pragma warning restore SYSLIB0014
-      webRequest.Method = WebRequestMethods.Http.Head;
-      webRequest.AllowAutoRedirect = true;
-      webRequest.Host = webApplicationRoot.Host;
-      webRequest.ServerCertificateValidationCallback += (sender, certificate, chain, errors) => true;
-
       HttpStatusCode statusCode = default;
       Assertion.DebugAssert(statusCode != HttpStatusCode.OK);
 
       while (statusCode != HttpStatusCode.OK)
       {
+  #pragma warning disable SYSLIB0014
+        var webRequest = (HttpWebRequest)HttpWebRequest.Create(resolvedUri); // TODO RM-8492: Replace with HttpClient
+  #pragma warning restore SYSLIB0014
+        webRequest.Method = WebRequestMethods.Http.Head;
+        webRequest.AllowAutoRedirect = true;
+        webRequest.Host = webApplicationRoot.GetComponents(UriComponents.Host | UriComponents.Port, UriFormat.UriEscaped);
+        webRequest.ServerCertificateValidationCallback += (sender, certificate, chain, errors) => true;
+
         try
         {
           var remainingTimeout = (int)(applicationPingTimeout.TotalMilliseconds - stopwatch.Elapsed.TotalMilliseconds);
-          webRequest.Timeout = Math.Max(remainingTimeout, 0);
-
-          using (var response = (HttpWebResponse)webRequest.GetResponse())
+          if (remainingTimeout > 0)
           {
-            statusCode = response.StatusCode;
+            webRequest.Timeout = remainingTimeout;
+
+            using (var response = (HttpWebResponse)webRequest.GetResponse())
+            {
+              statusCode = response.StatusCode;
+            }
           }
         }
         catch (WebException ex)
         {
+          if (ex.Response is HttpWebResponse httpWebResponse)
```

Comment:
Is there a valid scenario where this can actually happen or is it just to prevent infinite loops?

Reply:
Oh, yeah, now I see it.